src/components/CheckEditor/CheckEditor.test.tsx

```diff
@@ -311,6 +313,18 @@ describe('HTTP', () => {
     await act(async () => await userEvent.paste(screen.getByLabelText('Client Key', { exact: false }), 'client key'));
     await toggleSection('TLS config');
 
+    // Authentication
+    const authentication = await toggleSection('Authentication');
+    userEvent.click(await within(authentication).findByLabelText('Include bearer authorization header in request'));
+    const bearerTokenInput = await screen.findByPlaceholderText('Bearer token');
+    await act(async () => await userEvent.type(bearerTokenInput, 'a bearer token'));
+
+    userEvent.click(await within(authentication).findByLabelText('Include basic authorization header in request'));
+    const usernameInput = await within(authentication).findByPlaceholderText('Username');
+    const passwordInput = await within(authentication).findByPlaceholderText('Password');
+    await act(async () => await userEvent.type(usernameInput, 'a username'));
+    await act(async () => await userEvent.type(passwordInput, 'a password'));
```

Comment:
Why exactly do we need the `await`s and the `act()` here?

Reply:
It's because of react-hook-form. I think the validation functions are what do it, they are async by default, so any time you have input into a validated field React will throw `act` warnings in the tests. There might be additional reasons, but I'm not sure